### ISSUE TITLE
pop-pl overwrites the default filename extension in DrRacket

### ISSUE THREAD

**Issue by gregid** [florence/pop-pl#5]
After installing pop-pl the default extensions for definitions files in DrRacket has changed to *.pop.
In order to have standard #lang racket saved in *.rkt I need to provide full file name: NewName.rkt. If I give only NewName it will be saved as NewName.pop

**Comment by florence** [comment]
This is probably a bug with `module-suffixes`, which is the intended way for languages to add a new file extension.

I'm looking in to it

**Comment by florence** [comment]
What version of Racket are you on?

**Comment by rfindler** [comment]
This needs better support from DrRacket to fix.

**Comment by florence** [comment]
Moved to https://github.com/racket/drracket/issues/26

**Comment by rfindler** [comment]
@gregid what OS are you running?

**Comment by gregid** [comment]
I use Windows 10 and Racket 6.3

**Comment by rfindler** [comment]
@florence there is something language-specific already, actually. I had forgotten. 

http://docs.racket-lang.org/tools/adding-languages.html#%28part.__lang-based_.Languages_in_.Dr.Racket%29

**Comment by rfindler** [comment]
I think that POP-PL needs to support this protocol:

http://docs.racket-lang.org/tools/adding-languages.html#%28idx._%28gentag._15._%28lib._scribblings%2Ftools%2Ftools..scrbl%29%29%29

**Comment by florence** [comment]
I implemented that protocol however finder still defaults to ".rkt" when I try to save new pop-pl files. Did I mess up the get-info?

```
(define (get-info . default)
  (match-lambda**
   [('color-lexer _) lex]
   [('drracket:default-filters _)
    ".pop"]
   [(_ d) d]))
```

**Comment by rfindler** [comment]
I'm not sure, but this program doesn't see it, which is what needs to work. Maybe that's the wrong get-info function?

```
#lang racket/base
(define f (read-language (open-input-string "#lang pop-pl")))
(f 'drracket:default-filters #f)
```

**Comment by florence** [comment]
That program evals to `".pop"` on my machine. (I never pushed the new `get-info` function so of course you wouldn't see it).

**Comment by florence** [comment]
Oh I see whats going on, there was an error pop-up I missed. The documentation claims: 

`When a language’s get-info procedure responds to 'drracket:default-extension, it is expected to return (and/c string? (not/c #rx"[.]"))...` 

but the error i have says:

```
...e-language-tools.rkt:223:12: broke its own contract
  promised: (or/c #f (listof (list/c string? string?)))
  produced: ".pop"
  in: (or/c #f (listof (list/c string? string?)))
  contract from: #lang pop-pl
  blaming: #lang pop-pl
   (assuming the contract is correct)
  context...
```

**Comment by rfindler** [comment]
Can you please send the context too?​

**Comment by florence** [comment]
```
...e-language-tools.rkt:223:12: broke its own contract
  promised: (or/c #f (listof (list/c string? string?)))
  produced: ".pop"
  in: (or/c #f (listof (list/c string? string?)))
  contract from: #lang pop-pl
  blaming: #lang pop-pl
   (assuming the contract is correct)
  context...:
   /Applications/Racket/racket/collects/racket/contract/private/blame.rkt:156:0: raise-blame-error16
   /Users/florence/playground/drracket/drracket/drracket/private/module-language-tools.rkt:175:6: move-to-new-language method in ...e-language-tools.rkt:112:4
   /Applications/Racket/racket/share/pkgs/gui-lib/framework/private/logging-timer.rkt:41:0: log-timeline/proc
   /Applications/Racket/racket/share/pkgs/gui-lib/mred/private/wx/common/timer.rkt:34:38
   /Applications/Racket/racket/share/pkgs/gui-lib/mred/private/wx/common/queue.rkt:454:6
   /Applications/Racket/racket/share/pkgs/gui-lib/mred/private/wx/common/queue.rkt:505:32
   /Applications/Racket/racket/share/pkgs/gui-lib/mred/private/wx/common/queue.rkt:653:3
```

**Comment by rfindler** [comment]
Is it possible that you're showing me the error for a bad value
for 'drracket:default-filters and not 'drracket:default-extension?

**Comment by florence** [comment]
Yuuup, that was it. Fixed by d869da8
